pkg/controller/account/iam.go

```diff
@@ -358,3 +361,79 @@ func getSigninToken(reqLogger logr.Logger, federationEndpointURL string, federat
 	return signinResponse, nil
 
 }
+
+func (r *ReconcileAccount) smartCreateAccessKey(reqLogger logr.Logger, client awsclient.Client, userName string, account *awsv1alpha1.Account) (*iam.CreateAccessKeyOutput, error) {
```

Comment:
Could we rename this function `upsertAccessKey` I feel that makes sense here since its more descriptive of whats going to happen, update or insert (create)